### DISCUSSION TITLE
PorePy types

### DISCUSSION THREAD

**Discussion by vlipovac** [pmgbergen/porepy#731] [Ideas]
Hi,

in several places in the package there are definitions of Type Annotations like
`GridLike =  Union[pp.Grid, pp.MortarGrid]`
or
`VarLike =  Union[pp.ad.Variable, pp.ad.MergedVariable]` .

Is there some motivation (and would it make sense) to formally introduce PorePy types and define them clearly in a submodule for a uniform and consistent usage throughout the package?

BR,
Veljko

**Comment by IvarStefansson** [comment]
Yes, that is something we have considered and would totally make sense.

**Comment by keileg** [comment]
I second Ivar's opinion. It is not clear to me what is the standard way of defining such types (separate module or something else), but I'm sure the internet has an answer to it.
